lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -578,6 +587,9 @@ export default function getAPIImplementation(): APIInterface {
                   session: SessionContainerInterface;
                   user: User;
               }
+            | {
+                  status: "USER_BANNED_ERROR";
```

Comment:
no need to modify the API interface as we already have SIGN_IN_NOT_ALLOWED. Instead, modify the recipe interface.